### PR TITLE
feature: set pref in options to change new tab opening locstion

### PR DESCRIPTION
**Pull request by ericawright**
@bwinton 

In add-on options page user can now toggle preference of opening new tabs at bottom or top. Top is set as default.

fixes: #322

### CODE REVIEW

**Comment by bwinton** [comment]
Are we planning on opening new background tabs (with cmd-click) above/below as part of this preference, too?  Should we?

File: verticaltabs.jsm [hunk 1]
```diff
@@ -101,19 +101,25 @@ function VerticalTabs(window, {newPayload, addPingStats, AppConstants, setDefaul
...
     this.inferFromText = this.window.ToolbarIconColor.inferFromText;
     let AppConstants = this.AppConstants;
     let window = this.window;
     let document = this.document;
+    try {
+      Services.prefs.getBoolPref('extensions.verticaltabs.opentabstop');
+    } catch (ex) {
+      if (ex.result === Components.results.NS_ERROR_UNEXPECTED) {
+        Services.prefs.setBoolPref('extensions.verticaltabs.opentabstop', true);
+      }
```
Comment (bwinton) [inline verticaltabs.jsm]: We _probably_ want an else block here that logs (or re-throws) the exception, just so if something goes wrong we don't silently lose the stack trace…

**Comment by ericawright** [comment]
I suppose we can. Just above or below the current tab right? Maybe another
feature for that one.

File: verticaltabs.jsm [hunk 2]
```diff
@@ -101,19 +101,25 @@ function VerticalTabs(window, {newPayload, addPingStats, AppConstants, setDefaul
 VerticalTabs.prototype = {
 
   init: function () {
-    this.BrowserOpenTab = this.window.BrowserOpenTab;
-    this.window.BrowserOpenTab = function () {
-      this.pushToTop = true;
-      this.window.openUILinkIn(this.window.BROWSER_NEW_TAB_URL, 'tab');
-      this.pushToTop = false;
```
Comment (bwinton) [inline verticaltabs.jsm]: So, I'm a little confused as to why we don't need to set `pushToTop` back to `false` anymore…
So, I tried removing the `pushToTop` variable entirely, and just using `Services.prefs.getBoolPref('extensions.verticaltabs.opentabstop')` on line 465, and it seems to work okay (in Firefox Release), so maybe we want to do that instead?
(We still seem to need the `BrowserOpenTab` function for some reason, though.)

Comment (ericawright) [inline verticaltabs.jsm]: oh, good. yup you're right. nope, don't need the `BrowserOpenTab` function. Maybe you left in the override code?

**Comment by bwinton** [comment]
Oh, I just remembered that we also want to add whether people have set this flag or not to our Telemetry packet at https://github.com/bwinton/VerticalTabs/blob/master/utils.js#L169 (and https://github.com/bwinton/VerticalTabs/blob/master/docs/metrics.md#data-collection-details)

File: bootstrap.js [hunk 3]
```diff
@@ -93,6 +93,17 @@ function startup(data, reason) {
...
+  try {
+    Services.prefs.getBoolPref('extensions.verticaltabs.opentabstop');
+  } catch (ex) {
+    if (ex.result === Components.results.NS_ERROR_UNEXPECTED) {
+      Services.prefs.setBoolPref('extensions.verticaltabs.opentabstop', true);
+    } else {
+      throw(ex);
+    }
+  }
+
```
Comment (bwinton) [inline bootstrap.js]: Should this go _in_ setDefaultPrefs?  Cause it's kinda setting a default pref…

File: bootstrap.js [hunk 4]
```diff
@@ -73,6 +73,17 @@ function setDefaultPrefs() {
       break;
     }
   }
+  
+  //set new tabs to open at top as default -- only if undefined
+  try {
+    Services.prefs.getBoolPref('extensions.verticaltabs.opentabstop');
+  } catch (ex) {
+    if (ex.result === Components.results.NS_ERROR_UNEXPECTED) {
+      Services.prefs.setBoolPref('extensions.verticaltabs.opentabstop', true);
```
Comment (bwinton) [inline bootstrap.js]: Okay, one last thing…
I think we might want to remove this in removeDefaultPrefs, too.

Comment (ericawright) [inline bootstrap.js]: not if you want the pref to stick around....

Comment (bwinton) [inline bootstrap.js]: Oh, right…

**Pull request merged**